src/core/thread/mle_router.cpp

```diff
@@ -2502,6 +2511,16 @@ void MleRouter::SetLeaderWeight(uint8_t aWeight)
     mLeaderWeight = aWeight;
 }
 
+uint32_t MleRouter::GetLeaderPartitionId(void) const
+{
+	return mLeaderPartitionId;
+}
+
+void MleRouter::SetLeaderPartitionId(uint32_t aPartitionId)
+{
+	mLeaderPartitionId = aPartitionId;
```

Comment:
Partition id is set only by leader when forming a thread network. If setting the partition id after leader joins the network, the assigned partition id will take affect when router/leader quits from the current partition and then form the next thread partition. Maybe we don't need to return error at once here.